lnwallet/channel.go

```diff
@@ -2426,28 +2426,29 @@ func (lc *LightningChannel) ForceClose() (*ForceCloseSummary, error) {
 	}, nil
 }
 
-// InitCooperativeClose initiates a cooperative closure of an active lightning
-// channel. This method should only be executed once all pending HTLCs (if any)
-// on the channel have been cleared/removed. Upon completion, the source
-// channel will shift into the "closing" state, which indicates that all
+// CreateCloseProposal is used by both the initiator and responder in a
+// cooperative channel close workflow to generate their proposed close
+// transactions.  This method should only be executed once all pending HTLCs
+// (if any) on the channel have been cleared/removed. Upon completion, the
+// source channel will shift into the "closing" state, which indicates that all
 // incoming/outgoing HTLC requests should be rejected. A signature for the
-// closing transaction, and the txid of the closing transaction are returned.
-// The initiator of the channel closure should then watch the blockchain for a
-// confirmation of the closing transaction before considering the channel
-// terminated. In the case of an unresponsive remote party, the initiator can
-// either choose to execute a force closure, or backoff for a period of time,
-// and retry the cooperative closure.
+// closing transaction is returned.  Both the initiator and responder in
+// channel closure workflow should then watch the blockchain for a confirmation
+// of the closing transaction before considering the channel terminated. In the
+// case of an unresponsive remote party, the initiator can either choose to
+// execute a force closure, or backoff for a period of time, and retry the
+// cooperative closure.
 //
 // TODO(roasbeef): caller should initiate signal to reject all incoming HTLCs,
 // settle any inflight.
-func (lc *LightningChannel) InitCooperativeClose() ([]byte, *chainhash.Hash, error) {
+func (lc *LightningChannel) CreateCloseProposal() (*btcec.Signature, error) {
```

Comment:
Hmm, I think this should take a target fee rate. I think it'll complement the work you're doing to introduce dynamic fees into the project. Also it's a pretty minor diff in that we'd only be introducing a new parameter.